pkg/ebpf/uprobes/attacher.go

```diff
@@ -131,16 +131,37 @@ func (r *AttachRule) getProbeOptions(probeID manager.ProbeIdentificationPair) (P
 	}, nil
 }
 
-// Validate checks whether the rule is valid, returns nil if it is, an error message otherwise
-func (r *AttachRule) Validate() error {
+// Validate checks whether the rule is valid and compatible with the given attacher config, returns nil if it is, an error message otherwise
+func (r *AttachRule) Validate(attacherConfig *AttacherConfig) error {
 	var result error
 
 	if r.Targets == 0 {
 		result = multierror.Append(result, errors.New("no targets specified"))
 	}
 
-	if r.canTarget(AttachToSharedLibraries) && r.LibraryNameRegex == nil {
-		result = multierror.Append(result, errors.New("no library name regex specified"))
+	if r.canTarget(AttachToSharedLibraries) {
+		if r.LibraryNameRegex == nil {
+			result = multierror.Append(result, errors.New("no library name regex specified"))
+		}
+
+		// A sanity check to ensure that the library name regex matches at least one library in the libset.
+		// The best way to do this would be to check that the language generated by the "regex" used in the eBPF
+		// program is a subset of the language generated by the regexes in the libset, but that's a hard problem,
+		// so we do a simple check: either the regex matches a library name in the libset, or the regex contains
+		// a substring that matches a library name in the libset.
+		matchesAtLeastOneLib := false
+		suffixes := sharedlibraries.LibsetToLibSuffixes[attacherConfig.SharedLibsLibset]
+		for _, libSuffix := range suffixes {
+			libSuffixWithExt := libSuffix + ".so"
+			if r.LibraryNameRegex.MatchString(libSuffixWithExt) || strings.Contains(r.LibraryNameRegex.String(), libSuffix) {
+				matchesAtLeastOneLib = true
+				break
+			}
+		}
+
+		if !matchesAtLeastOneLib {
+			result = multierror.Append(result, fmt.Errorf("library name regex %s does not match any library in libset %s (%s)", r.LibraryNameRegex, attacherConfig.SharedLibsLibset, suffixes))
+		}
```

Comment:
I do recommend to add a test with r.LibraryNameRegex nil, as on line 156 you're accessing it, while it can be nil

Reply:
Good catch :D